### PR TITLE
chore(skore)!: Impose to pass `X_test` to EstimatorReport

### LINKED ISSUE #2627: make test data mandatory?
### What would you like to say?

ATM a report can be created without X_test and y_test. I get that training data is optional when we pass a fitted estimator, but maybe we should force a test set to be provided?

### PR DESCRIPTION
**Pull request by glemaitre**
closes #2627

Enforce to have some testing data when creating an `EstimatorReport`.

### CODE REVIEW

File: skore/src/skore/_sklearn/_comparison/report.py [hunk 1]
```diff
@@ -390,7 +390,7 @@ def create_estimator_report(
         self,
         *,
         report_key: str,
-        X_test: ArrayLike | None = None,
+        X_test: ArrayLike,
         y_test: ArrayLike | None = None,
```
Comment (GaetandeCast) [inline skore/src/skore/_sklearn/_comparison/report.py:394]: Since we dropped support for unsupervised learning, we should also expect `y_test` no?

Comment (glemaitre) [reply skore/src/skore/_sklearn/_comparison/report.py:394]: I would do in another PR then because we would need to clean up at several places. I think that we still detect clustering in some way but we don't have nice support for it.

**Comment by glemaitre** [comment]
I broke the `skore-mlflow-project` I think :)

File: skore/src/skore/_sklearn/_estimator/report.py
Comment (thomass-dev) [inline skore/src/skore/_sklearn/_estimator/report.py]: Why not leaving this parameter as positional, without default value?

Comment (thomass-dev) [reply skore/src/skore/_sklearn/_estimator/report.py]: It's counterintuitive that the parameter is mandatory but optional in the signature.

Comment (glemaitre) [reply skore/src/skore/_sklearn/_estimator/report.py]: Since we are keyword only, then we can do that indeed.

(still used to the all time where we don't use `*` and you could not have "positional" argument after one specifying a default).

File: skore/src/skore/_sklearn/_comparison/report.py (hunk 1 above)
Comment (glemaitre) [reply skore/src/skore/_sklearn/_comparison/report.py:394]: I'll open an issue now and affect myself to this cleanup.

File: skore-mlflow-project/src/skore_mlflow_project/reports.py [hunk 2]
```diff
@@ -199,8 +199,7 @@ def iter_cv(report: CrossValidationReport) -> Generator[NestedLogItem, None, Non
     """Yield loggable objects for a cross-validation report."""
     yield from iter_cv_metrics(report)
 
-    estimator_report = report.create_estimator_report()
-    estimator = estimator_report.estimator_
+    estimator = clone(report.estimator).fit(report.X, report.y)
```
Comment (glemaitre) [inline skore-mlflow-project/src/skore_mlflow_project/reports.py:202]: @thomass-dev this change is backward compatible but we touch the `skore-mlflow-project`.
